sample/src/main/java/com/airwallex/paymentacceptance/ui/UIIntegrationActivity.kt

```diff
@@ -0,0 +1,125 @@
+            }
+            mViewModel.updateCheckoutModel(selectedOption)
+        }
+        mBinding.btnPaymentList.setOnClickListener {
+            setLoadingProgress(true)
+            //launch the payment list page
+            mViewModel.launchPaymentList(this)
+        }
+        mBinding.btnCardPayment.setOnClickListener {
+            setLoadingProgress(true)
+            //launch the card payment page
+            mViewModel.launchCardPage(this)
+        }
+        mBinding.btnCardDialogPayment.setOnClickListener {
+            setLoadingProgress(true)
+            //launch the card payment dialog
+            mViewModel.launchCardDialog(this)
+        }
+        mBinding.btnShipping.setOnClickListener {
+            //launch the shipping page
+            mViewModel.launchShipping(this)
+        }
+        mBinding.imSetting.setOnClickListener {
+            startActivity(Intent(this, PaymentSettingsActivity::class.java))
+            overridePendingTransition(R.anim.slide_in_right, R.anim.slide_out_left)
+        }
+    }
+
+    override fun addObserver() {
+        mViewModel.airwallexPaymentStatus.observe(this) { status ->
+            handleStatusUpdate(status)
+        }
+        mViewModel.dialogShowed.observe(this) {
+            setLoadingProgress(false)
+        }
+        mViewModel.airwallexShippingStatus.observe(this) {
+
+        }
+        mViewModel.createPaymentIntentError.observe(this) { error ->
+            setLoadingProgress(false)
+            showAlert(
+                getString(R.string.create_payment_intent_failed),
+                error ?: getString(R.string.payment_failed_message)
+            )
+        }
+    }
+
+    private fun handleStatusUpdate(status: AirwallexPaymentStatus) {
+        setLoadingProgress(false)
+        when (status) {
+            is AirwallexPaymentStatus.Success -> {
+                Log.d(TAG, "Payment success ${status.paymentIntentId}")
+                showPaymentSuccess()
+            }
+
+            is AirwallexPaymentStatus.InProgress -> {
+                // redirecting
+                Log.d(TAG, "Payment is redirecting ${status.paymentIntentId}")
+                showPaymentInProgress()
+            }
+
+            is AirwallexPaymentStatus.Failure -> {
+                showPaymentError(status.exception.localizedMessage)
+            }
+
+            is AirwallexPaymentStatus.Cancel -> {
+                Log.d(TAG, "User cancel the payment")
+                showPaymentCancelled()
+            }
+        }
+    }
+
+    override fun onNewIntent(intent: Intent) {
+        super.onNewIntent(intent)
+
+        if (intent.scheme == AIRWALLEX_CHECKOUT_SCHEMA) {
+            showAlert(
+                getString(R.string.payment_successful),
+                getString(R.string.payment_successful_message)
+            )
+        }
+    }
+
+    companion object {
+        private const val TAG = "UIIntegrationActivity"
+
+        fun startActivity(context: Context) {
```

Comment:
Can this be written as a extension method for context like:
`fun <T: Any> Context.startActivity(clazz: KClass<T>) {
    startActivity(Intent(this, clazz::class.java))
}`

Reply:
OK